// @arco-design/web-vue/es/date-picker/index.js
import { setGlobalConfig, getComponentPrefix } from "@arco-design/web-vue/es/_utils/global-config.jstils/global-config.js";
import _DatePicker from "@arco-design/web-vue/es/date-picker/pickers/date-picker.jsdate-picker/pickers/date-picker.js";
import WeekPicker from "@arco-design/web-vue/es/date-picker/pickers/week-picker.jsdate-picker/pickers/week-picker.js";
export { default as WeekPicker } from "@arco-design/web-vue/es/date-picker/pickers/week-picker.jsdate-picker/pickers/week-picker.js";
import MonthPicker from "@arco-design/web-vue/es/date-picker/pickers/month-picker.jsate-picker/pickers/month-picker.js";
export { default as MonthPicker } from "@arco-design/web-vue/es/date-picker/pickers/month-picker.jsate-picker/pickers/month-picker.js";
import YearPicker from "@arco-design/web-vue/es/date-picker/pickers/year-picker.jsdate-picker/pickers/year-picker.js";
export { default as YearPicker } from "@arco-design/web-vue/es/date-picker/pickers/year-picker.jsdate-picker/pickers/year-picker.js";
import QuarterPicker from "@arco-design/web-vue/es/date-picker/pickers/quarter-picker.jse-picker/pickers/quarter-picker.js";
export { default as QuarterPicker } from "@arco-design/web-vue/es/date-picker/pickers/quarter-picker.jse-picker/pickers/quarter-picker.js";
import RangePicker from "@arco-design/web-vue/es/date-picker/range-picker.jsvue/es/date-picker/range-picker.js";
export { default as RangePicker } from "@arco-design/web-vue/es/date-picker/range-picker.jsvue/es/date-picker/range-picker.js";
const DatePicker = Object.assign(_DatePicker, {
  WeekPicker,
  MonthPicker,
  YearPicker,
  QuarterPicker,
  RangePicker,
  install: (app, options) => {
    setGlobalConfig(app, options);
    const componentPrefix = getComponentPrefix(options);
    app.component(componentPrefix + _DatePicker.name, _DatePicker);
    app.component(componentPrefix + YearPicker.name, YearPicker);
    app.component(componentPrefix + QuarterPicker.name, QuarterPicker);
    app.component(componentPrefix + MonthPicker.name, MonthPicker);
    app.component(componentPrefix + WeekPicker.name, WeekPicker);
    app.component(componentPrefix + RangePicker.name, RangePicker);
  }
});
export { DatePicker as default };
