// @arco-design/web-vue/es/image/index.js
import { setGlobalConfig, getComponentPrefix } from "@arco-design/web-vue/es/_utils/global-config.jstils/global-config.js";
import _Image from "@arco-design/web-vue/es/image/image.jsgn/web-vue/es/image/image.js";
import ImagePreview from "@arco-design/web-vue/es/image/preview.js/web-vue/es/image/preview.js";
export { default as ImagePreview } from "@arco-design/web-vue/es/image/preview.js/web-vue/es/image/preview.js";
import ImagePreviewGroup from "@arco-design/web-vue/es/image/preview-group.jsue/es/image/preview-group.js";
export { default as ImagePreviewGroup } from "@arco-design/web-vue/es/image/preview-group.jsue/es/image/preview-group.js";
import ImagePreviewAction from "@arco-design/web-vue/es/image/preview-action.jse/es/image/preview-action.js";
export { default as ImagePreviewAction } from "@arco-design/web-vue/es/image/preview-action.jse/es/image/preview-action.js";
const Image = Object.assign(_Image, {
  Preview: ImagePreview,
  PreviewGroup: ImagePreviewGroup,
  install: (app, options) => {
    setGlobalConfig(app, options);
    const componentPrefix = getComponentPrefix(options);
    app.component(componentPrefix + _Image.name, _Image);
    app.component(componentPrefix + ImagePreview.name, ImagePreview);
    app.component(
      componentPrefix + ImagePreviewGroup.name,
      ImagePreviewGroup
    );
    app.component(
      componentPrefix + ImagePreviewAction.name,
      ImagePreviewAction
    );
  }
});
export { Image as default };
