// @arco-design/web-vue/es/table/index.js
import { setGlobalConfig, getComponentPrefix } from "@arco-design/web-vue/es/_utils/global-config.jstils/global-config.js";
import _Table from "@arco-design/web-vue/es/table/table.jsgn/web-vue/es/table/table.js";
import Thead from "@arco-design/web-vue/es/table/table-thead.js-vue/es/table/table-thead.js";
export { default as Thead } from "@arco-design/web-vue/es/table/table-thead.js-vue/es/table/table-thead.js";
import Tbody from "@arco-design/web-vue/es/table/table-tbody.js-vue/es/table/table-tbody.js";
export { default as Tbody } from "@arco-design/web-vue/es/table/table-tbody.js-vue/es/table/table-tbody.js";
import Tr from "@arco-design/web-vue/es/table/table-tr.jsweb-vue/es/table/table-tr.js";
export { default as Tr } from "@arco-design/web-vue/es/table/table-tr.jsweb-vue/es/table/table-tr.js";
import Th from "@arco-design/web-vue/es/table/table-th.jsweb-vue/es/table/table-th.js";
export { default as Th } from "@arco-design/web-vue/es/table/table-th.jsweb-vue/es/table/table-th.js";
import Td from "@arco-design/web-vue/es/table/table-td.jsweb-vue/es/table/table-td.js";
export { default as Td } from "@arco-design/web-vue/es/table/table-td.jsweb-vue/es/table/table-td.js";
import TableColumn from "@arco-design/web-vue/es/table/table-column.jsvue/es/table/table-column.js";
export { default as TableColumn } from "@arco-design/web-vue/es/table/table-column.jsvue/es/table/table-column.js";
const Table = Object.assign(_Table, {
  Thead,
  Tbody,
  Tr,
  Th,
  Td,
  Column: TableColumn,
  install: (app, options) => {
    setGlobalConfig(app, options);
    const componentPrefix = getComponentPrefix(options);
    app.component(componentPrefix + _Table.name, _Table);
    app.component(componentPrefix + Thead.name, Thead);
    app.component(componentPrefix + Tbody.name, Tbody);
    app.component(componentPrefix + Tr.name, Tr);
    app.component(componentPrefix + Th.name, Th);
    app.component(componentPrefix + Td.name, Td);
    app.component(componentPrefix + TableColumn.name, TableColumn);
  }
});
export { Table as default };
