// @arco-design/web-vue/es/menu/index.js
import { setGlobalConfig, getComponentPrefix } from "@arco-design/web-vue/es/_utils/global-config.jstils/global-config.js";
import _Menu from "@arco-design/web-vue/es/menu/menu.jsign/web-vue/es/menu/menu.js";
import MenuItem from "@arco-design/web-vue/es/menu/item.jsign/web-vue/es/menu/item.js";
export { default as MenuItem } from "@arco-design/web-vue/es/menu/item.jsign/web-vue/es/menu/item.js";
import MenuItemGroup from "@arco-design/web-vue/es/menu/item-group.jsb-vue/es/menu/item-group.js";
export { default as MenuItemGroup } from "@arco-design/web-vue/es/menu/item-group.jsb-vue/es/menu/item-group.js";
import SubMenu from "@arco-design/web-vue/es/menu/sub-menu.jsweb-vue/es/menu/sub-menu.js";
export { default as SubMenu } from "@arco-design/web-vue/es/menu/sub-menu.jsweb-vue/es/menu/sub-menu.js";
const Menu = Object.assign(_Menu, {
  Item: MenuItem,
  ItemGroup: MenuItemGroup,
  SubMenu,
  install: (app, options) => {
    setGlobalConfig(app, options);
    const componentPrefix = getComponentPrefix(options);
    app.component(componentPrefix + _Menu.name, _Menu);
    app.component(componentPrefix + MenuItem.name, MenuItem);
    app.component(componentPrefix + MenuItemGroup.name, MenuItemGroup);
    app.component(componentPrefix + SubMenu.name, SubMenu);
  }
});
export { Menu as default };
