// @arco-design/web-vue/es/dropdown/index.js
import { setGlobalConfig, getComponentPrefix } from "@arco-design/web-vue/es/_utils/global-config.jstils/global-config.js";
import _Dropdown from "@arco-design/web-vue/es/dropdown/dropdown.jsweb-vue/es/dropdown/dropdown.js";
import Doption from "@arco-design/web-vue/es/dropdown/dropdown-option.js/es/dropdown/dropdown-option.js";
export { default as Doption } from "@arco-design/web-vue/es/dropdown/dropdown-option.js/es/dropdown/dropdown-option.js";
import Dgroup from "@arco-design/web-vue/es/dropdown/dropdown-group.jse/es/dropdown/dropdown-group.js";
export { default as Dgroup } from "@arco-design/web-vue/es/dropdown/dropdown-group.jse/es/dropdown/dropdown-group.js";
import Dsubmenu from "@arco-design/web-vue/es/dropdown/dropdown-submenu.jses/dropdown/dropdown-submenu.js";
export { default as Dsubmenu } from "@arco-design/web-vue/es/dropdown/dropdown-submenu.jses/dropdown/dropdown-submenu.js";
import DropdownButton from "@arco-design/web-vue/es/dropdown/dropdown-button.js/es/dropdown/dropdown-button.js";
export { default as DropdownButton } from "@arco-design/web-vue/es/dropdown/dropdown-button.js/es/dropdown/dropdown-button.js";
const Dropdown = Object.assign(_Dropdown, {
  Option: Doption,
  Group: Dgroup,
  Submenu: Dsubmenu,
  Button: DropdownButton,
  install: (app, options) => {
    setGlobalConfig(app, options);
    const componentPrefix = getComponentPrefix(options);
    app.component(componentPrefix + _Dropdown.name, _Dropdown);
    app.component(componentPrefix + Doption.name, Doption);
    app.component(componentPrefix + Dgroup.name, Dgroup);
    app.component(componentPrefix + Dsubmenu.name, Dsubmenu);
    app.component(componentPrefix + DropdownButton.name, DropdownButton);
  }
});
export { Dropdown as default };
