// @arco-design/web-vue/es/input/index.js
import { setGlobalConfig, getComponentPrefix } from "@arco-design/web-vue/es/_utils/global-config.jstils/global-config.js";
import _Input from "@arco-design/web-vue/es/input/input.jsgn/web-vue/es/input/input.js";
import InputSearch from "@arco-design/web-vue/es/input/input-search.jsvue/es/input/input-search.js";
export { default as InputSearch } from "@arco-design/web-vue/es/input/input-search.jsvue/es/input/input-search.js";
import InputPassword from "@arco-design/web-vue/es/input/input-password.jse/es/input/input-password.js";
export { default as InputPassword } from "@arco-design/web-vue/es/input/input-password.jse/es/input/input-password.js";
import InputGroup from "@arco-design/web-vue/es/input/input-group.js-vue/es/input/input-group.js";
export { default as InputGroup } from "@arco-design/web-vue/es/input/input-group.js-vue/es/input/input-group.js";
const Input = Object.assign(_Input, {
  Search: InputSearch,
  Password: InputPassword,
  Group: InputGroup,
  install: (app, options) => {
    setGlobalConfig(app, options);
    const componentPrefix = getComponentPrefix(options);
    app.component(componentPrefix + _Input.name, _Input);
    app.component(componentPrefix + InputGroup.name, InputGroup);
    app.component(componentPrefix + InputSearch.name, InputSearch);
    app.component(componentPrefix + InputPassword.name, InputPassword);
  }
});
export { Input as default };
